### DISCUSSION TITLE
Can Simpleperf be used to capture and present static kernel trace events?

### DISCUSSION THREAD

**Discussion by avri-altman-sndk** [android/ndk#1659] [Q&A] [answered]
Hi,
On galaxy S21, I am trying to record some storage trace events while running a storage benchmark app in the background.
the kernel version is: 
t2s:/data/local/tmp # cat /proc/version
Linux version 5.4.61-g2d6d87a5b (avri@BXYGM33) (Android (6443078 based on r383902) clang version 11.0.1 (https://android.googlesource.com/toolchain/llvm-project b397f81060ce6d701042b782172ed13bee898b79), LLD 11.0.1 (/buildbot/tmp/tmp6_m7QH b397f81060ce6d701042b782172ed13bee898b79)) #2 SMP PREEMPT Sun Jan 23 00:39:01 IST 2022
t2s:/data/local/tmp #

I am interested in some specific trace events in the ufs driver:
1|t2s:/data/local/tmp # simpleperf list | grep ufs
  ufs:ufshcd_auto_bkops_state
  ufs:ufshcd_clk_gating
  ufs:ufshcd_clk_scaling
  ufs:ufshcd_command
  ufs:ufshcd_init
  ufs:ufshcd_profile_clk_gating
  ufs:ufshcd_profile_clk_scaling
  ufs:ufshcd_profile_hibern8
  ufs:ufshcd_runtime_resume
  ufs:ufshcd_runtime_suspend
  ufs:ufshcd_system_resume
  ufs:ufshcd_system_suspend
  ufs:ufshcd_tm
  ufs:ufshcd_uic_command
  ufs:ufshcd_upiu
  ufs:ufshcd_upiu_no_data
  ufs_exynos_perf:ufs_exynos_perf_issue
  ufs_exynos_perf:ufs_exynos_perf_lock
  ufs_exynos_perf:ufs_exynos_perf_stat
t2s:/data/local/tmp #

Start recording some specific events:
t2s:/data/local/tmp # simpleperf record -e ufs:ufshcd_command,ufs:ufshcd_upiu,ufs:ufshcd_upiu_no_data -a --exclude-perf

whilst starting my Benchmarking app (AndroBench).  When the benchmark finishes - I stop recording: Ctrl + C.
a binary file perf.data exists now in data/local/tmp.
How can I dump all the parsed events - similar to "perf script" or as if I was doing #cat /sys/kernel/tracing/trace_pipe?
Thanks,
Avri

**Comment by stephenhines** [comment, marked as answer]
@yabinc would probably be the most knowledgeable one here.

**Reply by yabinc** [reply]
You can pull perf.data on host, and use report_sample.py --show_tracing_data option to dump tracing fields for each sample, as in https://android.googlesource.com/platform/prebuilts/simpleperf/+/refs/heads/master/report_sample.py#93.
It should also be easy to modify report_sample.py to show data as you like.
report_sample.py can be found in ndk release, or downloaded from https://android.googlesource.com/platform/prebuilts/simpleperf/.
